namecheap/namecheap_domain_record.go

```diff
@@ -0,0 +1,375 @@
+			},
+			"email_type": {
+				ConflictsWith: []string{"nameservers"},
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringInSlice([]string{"NONE", "FWD", "MXE", "MX", "OX"}, false),
+				Description:   "Possible values: NONE, FWD, MXE, MX, OX",
+			},
+			"mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "MERGE",
+				ValidateFunc: validation.StringInSlice([]string{"MERGE", "OVERWRITE"}, false),
+				Description:  "Possible values: MERGE (default), OVERWRITE",
+			},
+			"record": {
+				ConflictsWith: []string{"nameservers"},
+				Type:          schema.TypeSet,
+				Optional:      true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"hostname": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Sub-domain/hostname to create the record for",
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"A", "AAAA", "ALIAS", "CAA", "CNAME", "MX", "MXE", "NS", "TXT", "URL", "URL301", "FRAME"}, false),
+							Description:  "Possible values: A, AAAA, ALIAS, CAA, CNAME, MX, MXE, NS, TXT, URL, URL301, FRAME",
+						},
+						"address": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Possible values are URL or IP address. The value for this parameter is based on record type",
+						},
+						"mx_pref": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     10,
+							Description: "MX preference for host. Applicable for MX records only",
+						},
+						"ttl": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     1799,
+							Description: "Time to live for all record types. Possible values: any value between 60 to 60000",
+						},
+					},
+				},
+			},
+			"nameservers": {
+				ConflictsWith: []string{"email_type", "record"},
+				Type:          schema.TypeList,
+				Optional:      true,
+				MinItems:      1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+		},
+	}
+}
+
+func resourceRecordCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*namecheap.Client)
+
+	domain := data.Get("domain").(string)
+	mode := data.Get("mode").(string)
+	data.SetId(domain)
+
+	var emailType *string
+	var records []interface{}
+	var nameservers []interface{}
+
+	if emailTypeRaw, ok := data.GetOk("email_type"); ok {
+		emailTypeString := emailTypeRaw.(string)
+		emailType = &emailTypeString
+	}
+
+	if recordsRaw, ok := data.GetOk("record"); ok {
+		records = recordsRaw.(*schema.Set).List()
+	}
+
+	if nameserversRaw, ok := data.GetOk("nameservers"); ok {
+		nameservers = nameserversRaw.([]interface{})
+	}
+
+	if mode == "MERGE" && records != nil {
```

Comment:
Do we want to use `strings.EqualFold` instead of case sensitive equality?

Reply:
I thought about it to set it as non case sensitive.
The only reason is consistency, since other fields (e.g. record.type field or email_type) requires values in upper case, so that's why I leaved it in upper case.

However, from developer experience perspective, probably you're right.
If you think it is better, I will easily amend it

Reply:
@adamdecaf I have made a few changes. Now mode & records comparison is safer.
Thanks for the comments! :)